backend/tcs-integration/Controllers/SessionsController.test.cs

```diff
@@ -0,0 +1,165 @@
+        {
+            var client = _factory.CreateClient();
+            var user = await Login(client);
+            var request = new HttpRequestMessage(HttpMethod.Get, "api/sessions/signedin");
+            request.Headers.Add("Authorization", $"Bearer {user.Token}");
+
+            var response = await client.SendAsync(request);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var sessions = await response.Content.ReadAsAsync<IEnumerable<Session>>();
+
+            Assert.All(sessions, s => Assert.Null( s.OutTime ));
+        }
+
+        [Fact]
+        public async void POST_sessions_ShouldReturn201WithCreatedSession()
+        {
+            var client = _factory.CreateClient();
+            var session = new SessionCreateDTO()
+            {
+                InTime = DateTime.Now,
+                PersonId = 1,
+                Tutoring = true,
+                SelectedClasses = new List<Class>() { new Class { CRN = 4, DepartmentCode = 2, Name = "ENGLISH LIT", ShortName = "English 101" } } ,
+                SelectedReasons = new List<Reason>() { new Reason { Id = 2, Name = "Computer Use", Deleted = true} }  
+            };
+            var user = await Login(client);
+            var request = new HttpRequestMessage(HttpMethod.Post, "api/sessions");
+            request.Headers.Add("Authorization", $"Bearer {user.Token}");
+            request.Content = new StringContent(JsonConvert.SerializeObject(session));
+            request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+
+            var response = await client.SendAsync(request);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var createdSession = await response.Content.ReadAsAsync<Session>();
+            Assert.Equal(1, createdSession.PersonId );
+            Assert.True( session.Tutoring);
+        }
+
+        [Fact]
+        public async void PUT_sessions_id_ShouldReturn200WithUpdatedSession()
+        {
+            var client = _factory.CreateClient();
+            var user = await Login(client);
+            var request = new HttpRequestMessage(HttpMethod.Put, "api/sessions/1");
+            request.Headers.Add("Authorization", $"Bearer {user.Token}");
+            request.Content = new StringContent(JsonConvert.SerializeObject(new Session {ID = 1, PersonId = 5, Tutoring = false, InTime = DateTime.Now, OutTime = null }));
+            request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+
+            var response = await client.SendAsync(request);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            var session = await response.Content.ReadAsAsync<Session>();
+            Assert.False( session.Tutoring);
+            Assert.Equal(5, session.PersonId);
+        }
+
+        [Fact]
+        public async void PUT_signout_id_ShouldReturn200WithSignedOutSession()
```

Comment:
Add in tests for our error cases.
* Signing Out when you aren't signed in.
* Not Selected at least one selectedClass
* Not selected at least one selectedReason
* Being able to signin with tutoring true and at least once selectedReason.

Reply:
A student can sign in with: 
- No class selected and One (or more) reason selected (Ex.  No class selected but "Computer Use selected as Reason" )
- One (or more) class and no reason selected (assuming tutoring is true)  (Ex.  One class selected and tutoring=true but no other reason selected)

Both of the above are true?  

Being able to signin with tutoring true and at least once selectedReason. --  What does this mean exactly?

All of these issues apply to PostSession as well as PutSession, right?

Reply:
> Both of the above are true?

No, the first isn't the second is.

At least 1 class must always be selected.

At least 1 reason must be selected or tutoring=true to be valid.

> Being able to signin with tutoring true and at least once selectedReason. -- What does this mean exactly?
That was my bad, I meant selectedClass

> All of these issues apply to PostSession as well as PutSession, right?

Yes.